### PR TITLE
Remove bots from the contributors list

### LINKED ISSUE #5098: Filter bots from contributors list
Currently there are several bot accounts that appear in our contributors list. Three of the contributors listed on our contributors page are bots, and [renovate-bot](https://github.com/renovate-bot) appears in the top 10 contributors. 

To better recognize human contributors, we should filter out bots from the contributors list. 

We use [this endpoint](https://docs.github.com/en/rest/repos/repos?apiVersion=2022-11-28#list-repository-contributors) to fetch contributors. The current list of contributors can be seen [here](https://api.github.com/repos/canonical/vanilla-framework/contributors). Each user has a `type` attribute that can be `User` or `Bot`. We can filter out users with `type=='Bot'`. The exception is [renovate-bot](https://github.com/renovate-bot) which, while being a bot, is not treated like a bot by Github as it has `type='User'`.

### PR DESCRIPTION
**Pull request by Satishpokala124**
## Done

Removes bots from the contributors list

Fixes #5098 

## QA

- Open [contributors page](insert-demo-url)
- Check if bots are removed from the contributors list.

### Check if PR is ready for release

If this PR contains Vanilla SCSS code changes, it should contain the following changes to make sure it's ready for the release:

- [x] PR should have one of the following labels to automatically categorise it in release notes:
  - `Feature 🎁`, `Breaking Change 💣`, `Bug 🐛`, `Documentation 📝`, `Maintenance 🔨`.
- [ ] Vanilla version in `package.json` should be updated relative to the [most recent release](https://github.com/canonical/vanilla-framework/releases/latest), following semver convention:
  - if CSS class names are not changed it can be bugfix release (x.x.**X**)
  - if CSS class names are changed/added/removed it should be minor version (x.**X**.0)
  - see the [wiki for more details](https://github.com/canonical/vanilla-framework/wiki/Release-process#pre-release-tasks)
- [ ] Any changes to component class names (new patterns, variants, removed or added features) should be listed on the [what's new page](https://github.com/canonical/vanilla-framework/blob/main/releases.yml).

## Screenshots

image

### CODE REVIEW

**Comment by jmuzina** [comment]
Hi @Satishpokala124, thanks for the contribution! 

Could you update this to also filter out [renovate-bot](https://github.com/renovate-bot) please? It is currently the bot with the most contributions so I'd like to make sure that it is also filtered out by this change.

**Comment by Satishpokala124** [comment]
Hi @jmuzina, the [renovate-bot](https://github.com/renovate-bot) is removed using its login. Let me know if that's ok.

File: webapp/app.py [hunk 1]
```diff
@@ -178,17 +178,39 @@ def _get_contributors():
...
         for contributor in contributors
         if contributor["login"] not in member_usernames
     ]
 
 
+def _is_not_renovate_bot(contributor):
+    # Using "login" to check for renovate-bot as
+    # GitHub treats it as a user, i.e., type: "User"
+    
+    return contributor["login"] != "renovate-bot"
```
Comment (jmuzina) [inline webapp/app.py:194]: I'd rather not make a function specifically for the `renovate-bot`, and as far as I know this bot is the only special case we need to worry about currently. So for now, can you un-extract this renovate-specific function?

Comment (jmuzina) [reply webapp/app.py:194]: `login` can be changed, so this filtering is somewhat [brittle](https://docs.github.com/en/rest/users/users?apiVersion=2022-11-28#get-a-user-using-their-id) in the case that renovate decides to change its username. Can we use `id` of `renovate-bot` (found [here](https://api.github.com/users/renovate-bot)) instead?

File: webapp/app.py [hunk 2]
```diff
@@ -178,17 +178,39 @@ def _get_contributors():
...
+
+
+def _filter_bots_from_contributors(contributors):
+    return [
+        contributor
+        for contributor in contributors
+        if (
+            contributor["type"].lower() != "bot" 
+            and _is_not_renovate_bot(contributor)
```
Comment (jmuzina) [inline webapp/app.py:203]: We only have one case of this currently, so I don't think we need to create a whole extra function for special-case bots yet; how about something like this?

```suggestion
            and contributor["id"] != 25180681 # renovate-bot
```

File: webapp/app.py [hunk 3]
```diff
@@ -178,17 +178,39 @@ def _get_contributors():
     return contributors
 
 
-def _filter_contributors(contributors):
-    # Distinguish team_members from contributors
```
Comment (jmuzina) [inline webapp/app.py:182]: Can we keep this comment about distinguishing team_members in the code please?

**Review by jmuzina — CHANGES_REQUESTED** [review]
Left a few comments, but this is almost ready to go! :)

File: webapp/app.py (hunk 1 above)
Comment (Satishpokala124) [reply webapp/app.py]: Got it! Using `id` instead of `login`.

File: webapp/app.py (hunk 3 above)
Comment (Satishpokala124) [reply webapp/app.py:182]: Sure. I have added it back.

**Review by jmuzina — APPROVED** [review]
LGTM, thanks for the contribution!

@bartaz this is looking good & ready to merge

**Comment by bartaz** [comment]
Thanks @Satishpokala124 for the fix and @jmuzina for review!

**Pull request merged**